### PR TITLE
WIP: Providing more optimization for the apache camel springboot example

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
This PR adds further optimizations to the previous changes for the apache camel example

### What issues does this PR fix or reference?
Follow-up: https://github.com/eclipse/che-devfile-registry/pull/46
Issue: https://github.com/redhat-developer/che-functional-tests/issues/556

### CODE REVIEW

File: devfiles/apache-camel-springboot/devfile.yaml [hunk 1]
```diff
@@ -16,11 +16,19 @@ components:
   -
     type: chePlugin
     id: camel-tooling/vscode-apache-camel/latest
+<<<<<<< Updated upstream
```
Comment (ibuziuk) [inline devfiles/apache-camel-springboot/devfile.yaml]: @ScrewTSW format looks completely broken

**Comment by ibuziuk** [comment]
@ScrewTSW please explain why those changes will improve the workspace startup?

**Comment by ibuziuk** [comment]
@apupier since you are the primary maintainer of the camel devfile adding you as a reviewer ;-)

**Comment by ScrewTSW** [comment]
> @ScrewTSW please explain why those changes will improve the workspace startup / stability in comparrison with the current values ?

I've pretty much just reassigned the available resources based on the consumption that I've seen during the testing.

**Comment by ibuziuk** [comment]
@ScrewTSW could you please provide comparison of RAM consumption for the current version of devfile and the one from PR ? what are the benefits? If, as you said, current values work well why do we need further optimization?

**Comment by ScrewTSW** [comment]
Current image memory consumption:

static memory consumptions:
* JWT Proxy consumes 150MiB
* Machine exec pod consumes 150MiB
* Theia consumes **512MB** = 488.281MiB

and the plugins defined in the devfile:
* vscode-xml : 128MiB
* vscode-apache-camel : 200MiB
* java lsp : 1400 MiB
* maven : 512 MiB

That leaves me 43.719MiB to be redistributed.

The memory usage in the LSP peaks at 1250MiB during use and then falls back down as GC takes care of the objects.
Apache camel peaks at 150MiB
Apache XML peaks at 95MiB
and finally Maven with the deployment running and in use peaks around 380 MiB

If you think there's no need to redistribute the extra memory that we have at our disposal, then this PR can be closed.

**Comment by ibuziuk** [comment]
@ScrewTSW sorry, but I probably do not understand the math.

> That leaves me 43.719MiB to be redistributed.

 In the PR you:

- add 56 MB to vscode-apache-camel
- removes 120 MB from java plugin
- removes 28 MB form maven

As I understand your motivation was to redistribute 43.719MiB, but in this PR you literally decrease RAM on 92 MB, and the most worrying thing for me that the RAM for java plugin, which is really memory hungry, is reduced. Am I missing smth ?

**Comment by ScrewTSW** [comment]
> @ScrewTSW sorry, but I probably do not understand the math.
> 
> > That leaves me 43.719MiB to be redistributed.
> 
> In the PR you:
> 
>     * add 56 MB to vscode-apache-camel
> 
>     * removes 120 MB from java plugin
> 
>     * removes 28 MB form maven
> 
> 
> As I understand your motivation was to redistribute 43.719MiB, but in this PR you literally decrease RAM on 92 MB, and the most worrying thing for me that the RAM for java plugin, which is really memory hungry, is reduced. Am I missing smth ?

I have been using those values before I was notified by Angel that he was able to get OOM on the LSP when only using 1Gi of memory.

That's why I've sent my last comment with the analysis and the current status of the image. As I said there, this PR can be closed as irrelevant if you decide that you don't want me to do some magic with the 43Mi we have at our disposal.

**Comment by ibuziuk** [comment]
>  have been using those values before

@ScrewTSW you are using those values now in the current PR :-)
So, if you want some feedback on optimization - `bring this image to full 3Gi` please update the PR accordingly. If not - just close it

**Comment by ScrewTSW** [comment]
Closing as irrelevant

**Pull request closed without merging**